packages/core/src/external/carequality/pd/process-incoming-pd.ts

```diff
@@ -87,17 +38,20 @@ export async function processIncomingRequest(
   mpi: MPI
 ): Promise<PatientDiscoveryRespToExternalGW> {
   try {
-    const patient = validateFHIRAndExtractPatient(payload.patientResource);
+    const patient = validateFHIRAndExtractPatient(payload);
     const matchingPatient = await mpi.findMatchingPatient(patient);
     if (!matchingPatient) {
-      return constructNoMatchResponse(payload);
+      return constructPDNoMatchResponse(payload);
     }
     return constructMatchResponse(payload, patientMPIToPartialPatient(matchingPatient));
   } catch (error) {
     if (error instanceof IHEGatewayError) {
-      return constructErrorResponse(payload, error);
+      return constructPDErrorResponse(payload, error);
     } else {
-      return constructErrorResponse(payload, new XDSRegistryError("Internal Server Error", error));
+      return constructPDErrorResponse(
+        payload,
+        new XDSRegistryError("Internal Server Error", error)
+      );
     }
```

Comment:
This seem to be done exactly the same in multiple places, maybe have a shared function on the `carequality/error.ts` file?

Reply:
Its not that many LOCs and the abstraction needed to handle all of them IMO makes things more complicated